### PR TITLE
PyReleaseValidation: move DD4hep test workflow into upgrade special wf section

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

This PR is a follow-up of the discussion in #29975 about the possible workflow to test DD4hep in a standard environment. As now a 2021 full geometry is available, a list of possible workflows fully based on DD4hep can be easily added as suggested by @kpedro88. The workflows themselves are failing, and for this reason they are not made available to the IB or short matrix. But they can be used for routine checks.

As offset ".911" has been chosen, it does not overlaps with any other use to my knowledge.

#### PR validation:

```
17:41 cmsdev25 1039> runTheMatrix.py --what upgrade -l 11634.911 --dryRun 

ignoring non-requested file relval_standard
ignoring non-requested file relval_highstats
ignoring non-requested file relval_pileup
ignoring non-requested file relval_generator
ignoring non-requested file relval_extendedgen
ignoring non-requested file relval_production
ignoring non-requested file relval_ged
processing relval_upgrade
ignoring non-requested file relval_2017
ignoring non-requested file relval_2026
ignoring non-requested file relval_identity
ignoring non-requested file relval_machine
ignoring non-requested file relval_unsch
ignoring non-requested file relval_premix
Running in 4 thread(s)

Preparing to run 11634.911 TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021

# in: /build/fabiocos/112X/dd4hep-test/CMSSW_11_2_X_2020-05-26-2300/work dryRun for 'cd 11634.911_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021
 cmsDriver.py TTbar_14TeV_TuneCP5_cfi  --conditions auto:phase1_2021_realistic -n 10 --era Run3_dd4hep --eventcontent FEVTDEBUG --relval 9000,100 -s GEN,SIM --datatier GEN-SIM --beamspot Run3RoundOptics25ns13TeVLowSigmaZ --geometry DD4hepExtended2021 --fileout file:step1.root  > step1_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021.log  2>&1
 

# in: /build/fabiocos/112X/dd4hep-test/CMSSW_11_2_X_2020-05-26-2300/work dryRun for 'cd 11634.911_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021
 cmsDriver.py step2  --conditions auto:phase1_2021_realistic -s DIGI:pdigi_valid,L1,DIGI2RAW,HLT:@relval2021 --datatier GEN-SIM-DIGI-RAW -n 10 --geometry DD4hepExtended2021 --era Run3_dd4hep --eventcontent FEVTDEBUGHLT --filein  file:step1.root  --fileout file:step2.root  > step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021.log  2>&1
 

# in: /build/fabiocos/112X/dd4hep-test/CMSSW_11_2_X_2020-05-26-2300/work dryRun for 'cd 11634.911_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021
 cmsDriver.py step3  --conditions auto:phase1_2021_realistic -n 10 --era Run3_dd4hep --eventcontent RECOSIM,MINIAODSIM,DQM --runUnscheduled  -s RAW2DIGI,L1Reco,RECO,RECOSIM,EI,PAT,VALIDATION:@standardValidation+@miniAODValidation,DQM:@standardDQM+@ExtraHLT+@miniAODDQM --datatier GEN-SIM-RECO,MINIAODSIM,DQMIO --geometry DD4hepExtended2021 --filein  file:step2.root  --fileout file:step3.root  > step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021.log  2>&1
 

# in: /build/fabiocos/112X/dd4hep-test/CMSSW_11_2_X_2020-05-26-2300/work dryRun for 'cd 11634.911_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021
 cmsDriver.py step4  --conditions auto:phase1_2021_realistic -s HARVESTING:@standardValidation+@standardDQM+@ExtraHLT+@miniAODValidation+@miniAODDQM --scenario pp --filetype DQM --geometry DD4hepExtended2021 --era Run3_dd4hep --mc  -n 100  --filein file:step3_inDQM.root --fileout file:step4.root  > step4_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021.log  2>&1

# in: /build/fabiocos/112X/dd4hep-test/CMSSW_11_2_X_2020-05-26-2300/work dryRun for 'cd 11634.911_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021
 cmsDriver.py step5  --conditions auto:phase1_2021_realistic -n 10 --era Run3_dd4hep --eventcontent ALCARECO --filein file:step3.root -s ALCA:SiPixelCalSingleMuon+TkAlMuonIsolated+TkAlMinBias+MuAlOverlaps+EcalESAlign+TkAlZMuMu+HcalCalHBHEMuonFilter+TkAlUpsilonMuMu+TkAlJpsiMuMu+SiStripCalMinBias --datatier ALCARECO --geometry DD4hepExtended2021 --fileout file:step5.root  > step5_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021.log  2>&1
 
11634.911_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull_DD4hep+DigiFull_DD4hep_2021+RecoFull_DD4hep_2021+HARVESTFull_DD4hep_2021+ALCAFull_DD4hep_2021 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Wed May 27 17:40:48 2020-date Wed May 27 17:40:48 2020; exit: 0 0 0 0 0
1 1 1 1 1 tests passed, 0 0 0 0 0 failed
```

The list of workflows is just enlarged by the desired addition.

### CODE REVIEW

**Comment by fabiocos** [comment]
@bsunanda this might be of interest for you

**Comment by fabiocos** [comment]
please test

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -638,6 +638,24 @@ def workflow_(self, workflows, num, fragment, stepList):
     offset = 0.1001,
 )
 
+class UpgradeWorkflow_DD4hep(UpgradeWorkflow):
+    def setup_(self, step, stepName, stepDict, k, properties):
+        stepDict[stepName][k] = merge([{'--geometry': 'DD4hepExtended2021', '--era': 'Run3_dd4hep'}, stepDict[step][k]])
+    def condition(self, fragment, stepList, key, hasHarvest):
+        return '2021' in key
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:645]: do we need to add this for every gen fragment, or is ttbar sufficient? (in other words, i recommend adding `fragment=="TTbar_13"` as an additional condition)

Comment (fabiocos) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:645]: @kpedro88 well, for initial tests I believe that particle guns would be more suitable, TTbar is a usual reference in the longer term. We may specialize it if really needed, but is this really creating a problem?

Comment (kpedro88) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:645]: It just makes the upgrade matrix list of workflows very long.

Comment (fabiocos) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py]: well, in my check il looks like a few percent effect, anyway we may reduce to a subset of reference samples, TTbar_13, SingleMuPt10 and ZMM_13 which was chosen by @slomeo in the test recently added by @bsunanda

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
failures are clearly unrelated to this PR (which has no effect in this test, it requires ```--what upgrade``` to become active)

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
@chayanit any comment?

**Comment by chayanit** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**